### DISCUSSION TITLE
Connection reset by peer Issue

### DISCUSSION THREAD

**Discussion by mrabhishekchauhan** [emqx/emqx#8018] [Q&A]
Hey Team,

I am using emqx version 4.4.2 and running emqx as a cluster 0f 3 nodes in a kubernetes environment( each with 4vcpu and 8gb memory). I have subscribed to some topics in emqx via a microservice running 3 instances in 3 separate pods. I see that when the publish happens for the first time there is an error that occurs frequently i.e. "Connection reset by peer" due to the heavy load of the messages maybe and after some time it stabilizes. I am not able to debug as to why this error occurs in the beginning and what can i do to fix this? 

P.S. : I have also enabled the autoReconnect and setReconnecting properties but they also do not help.

**Comment by sliusardanya** [comment]
What are your "heavy load" environment numbers? 
Did you check this issue - https://github.com/emqx/emqx/issues/3848 ?

I am stuck with the same problem, and the only possible solution is to make a different queue before MQTT..

**Comment by lafirest** [comment]
@mrabhishekchauhan
 Have you tried adjusting the connection rate?
